File: golang/linked_list.go
package main

import "fmt"

type Node struct {
	value int
	next  *Node
}

func NewNode(value int) *Node {
	return &Node{
		value: value,
		next:  nil,
	}
}

type LinkedList struct {
	head, tail *Node
}

func NewList() *LinkedList {
	return &LinkedList{
		head: nil,
		tail: nil,
	}
}

func (self *LinkedList) Insert(value int) {
	if self.head == nil {
		self.head = NewNode(value)
		self.tail = self.head
		return
	}
	new_node := NewNode(value)
	var prev_ptr *Node = nil
	ptr := self.head
	for ptr != nil && value > ptr.value {
		prev_ptr = ptr
		ptr = ptr.next
	}
	if prev_ptr != nil {
		prev_ptr.next = new_node
	}
	new_node.next = ptr

	if prev_ptr == nil {
		self.head = new_node
	}
	if ptr == nil {
		self.tail = new_node
	}
}

func (self *LinkedList) Print() {
	ptr := self.head
	for ptr != nil {
		fmt.Print(ptr.value)
		ptr = ptr.next
	}
	fmt.Println()
}

func main() {
	fmt.Println("hello world!")

	l := NewList()
	l.Insert(1)
	l.Insert(0)
	l.Insert(3)
	l.Insert(2)
	l.Print()
}

File: node/LinkedList.js
class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}

class LinkedList {
    head = null;
    tail = null;

    insert(value) {
        if (this.head === null) {
            this.head = new Node(value);
            this.tail = this.head;
            return;
        }
        let newNode = new Node(value);
        this.tail.next = newNode;
        this.tail = newNode;
    }

    insert(value) {
        if (this.head === null) {
            this.head = new Node(value);
            this.tail = this.head;
            return;
        }
        let newNode = new Node(value);
        let prevPtr = null;
        let ptr = this.head;
        while (ptr !== null && value > ptr.value) {
            prevPtr = ptr;
            ptr = ptr.next;
        }
        if (prevPtr !== null) {
            prevPtr.next = newNode;
        }
        newNode.next = ptr;
        if (prevPtr === null)
            this.head = newNode;
        if (ptr === null)
            this.tail = newNode;
    }

    printList() {
        let ptr = this.head;
        while (ptr !== null) {
            process.stdout.write(ptr.value.toString());
            ptr = ptr.next;
        }
    }
}

const list = new LinkedList();
list.insert(1);
list.insert(0);
list.insert(3);
list.insert(2);
list.printList();